include/awkward/array/RawArray.h

```diff
@@ -920,6 +923,110 @@ namespace awkward {
       throw std::runtime_error("FIXME: RawArray:reduce_next");
     }
 
+    const ContentPtr
+      sort_next(int64_t negaxis,
+                const Index64& starts,
+                const Index64& parents,
+                int64_t outlength,
+                bool ascending,
+                bool stable,
+                bool keepdims) const override {
+      std::shared_ptr<T> ptr(
+                    new T[(size_t)length_], util::array_deleter<T>());
+      std::vector<size_t> result(length_);
+
+      std::iota(result.begin(), result.end(), 0);
+      if(ascending  &&  !stable) {
+        std::sort(result.begin(), result.end(),
+          [&](size_t i1, size_t i2) {return getitem_at_nowrap(i1) < getitem_at_nowrap(i2);});
+      }
+      else if(!ascending  &&  !stable) {
+        std::sort(result.begin(), result.end(),
+          [&](size_t i1, size_t i2) {return getitem_at_nowrap(i1) > getitem_at_nowrap(i2);});
+      }
+      else if(ascending  &&  stable) {
+        std::stable_sort(result.begin(), result.end(),
+          [&](size_t i1, size_t i2) {return getitem_at_nowrap(i1) < getitem_at_nowrap(i2);});
+      }
+      else if(!ascending  &&  stable) {
+        std::stable_sort(result.begin(), result.end(),
+          [&](size_t i1, size_t i2) {return getitem_at_nowrap(i1) > getitem_at_nowrap(i2);});
+      }
+      struct Error err = util::awkward_numpyarray_sort<T>(
+        ptr.get(),
+        ptr_.get(),
+        &result[0],
+        starts.ptr().get(),
+        parents.ptr().get(),
+        parents.offset(),
+        length_);
+      util::handle_error(err, classname(), nullptr);
```

Comment:
**HERE**

I see that the `result` is the container where the sorting happens; this should be an `Index64` if it must be an argsort-style index, and then you can use it in `carry` to apply that index to the `ptr`.

Reply:
done